src/backend/distributed/commands/trigger.c

```diff
@@ -111,19 +158,24 @@ GetTriggerNameById(Oid triggerId)
 													useIndex, NULL, scanKeyCount,
 													scanKey);
 
-	char *triggerName = NULL;
+	HeapTuple targetHeapTuple = NULL;
 
 	HeapTuple heapTuple = systable_getnext(scanDescriptor);
 	if (HeapTupleIsValid(heapTuple))
 	{
-		Form_pg_trigger triggerForm = (Form_pg_trigger) GETSTRUCT(heapTuple);
-		triggerName = pstrdup(NameStr(triggerForm->tgname));
+		targetHeapTuple = heap_copytuple(heapTuple);
 	}
 
 	systable_endscan(scanDescriptor);
 	heap_close(pgTrigger, NoLock);
 
-	return triggerName;
+	if (targetHeapTuple == NULL && missingOk == false)
```

Comment:
Minor: !missingOk

Also I think `errorOnMissing` is a better name than missingOk, because we basically error when it is missing.

Reply:
`errorOnMissing` would mean the opposite but in general we have `missingOk`, I think it would be a bit unusal api

Reply:
Yeah that is also right, I guess we can change them all in another PR maybe, because `missingOk` isn't explicit about what happens when it is missing. However `errorOnMissing` or something similar is pretty explicit.